src/transformers/generation_utils.py

```diff
@@ -747,6 +840,16 @@ def greedy_search(
 
             >>> print("Generated:", tokenizer.batch_decode(outputs, skip_special_tokens=True))
         """
+        logits = None
+        decoder_attentions = () if output_attentions else None
+        decoder_hidden_states = None
+
+        if "encoder_outputs" in model_kwargs:
```

Comment:
can we do if `self.is_encoder_decoder:` ? It's a nit but we always use the `is_encoder_decoder` flag to check whether the model is an encoder-decoder in `generate()`